core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -207,8 +191,11 @@ public void runInContext() {
       });
       return;
     }
-    prepareHeaders(headers, callOptions, userAgent,
-        knownMessageEncodingRegistry, decompressorRegistry, compressorRegistry);
+    if (compressor == null) {
```

Comment:
nm. Apparently you can't call setCompression multiple times, so it looks like you are using the `null` value.